libs/database/src/lib/services/user.service.v2.ts

```diff
@@ -209,4 +211,116 @@ export class UserServiceV2 {
 
     return pages as PageResponseDto[];
   }
+
+  async blockUser(user: User, targetCastcleId: string) {
+    const blockUser = await this.repositoryService
+      .findUser({
+        _id: targetCastcleId,
+      })
+      .exec();
+
+    if (!blockUser) throw CastcleException.USER_OR_PAGE_NOT_FOUND;
+
+    await Promise.all([
+      this.repositoryService.updateRelationship(
+        { user: user._id, followedUser: blockUser._id },
+        {
+          $setOnInsert: {
+            user: user._id,
+            followedUser: blockUser._id,
+            visibility: EntityVisibility.Publish,
+            following: false,
+            blocked: false,
+          },
+          $set: { blocking: true },
+        },
+        { upsert: true }
+      ),
+      this.repositoryService.updateRelationship(
+        { followedUser: user._id, user: blockUser._id },
+        {
+          $setOnInsert: {
+            followedUser: user._id,
+            user: blockUser._id,
+            visibility: EntityVisibility.Publish,
+            following: false,
+            blocking: false,
+          },
+          $set: { blocked: true },
+        },
+        { upsert: true }
+      ),
+    ]);
+  }
+
+  async getBlockedLookup(
+    user: User,
+    { hasRelationshipExpansion, maxResults, sinceId, untilId }: PaginationQuery
+  ) {
+    const filterQuery = {
+      sinceId,
+      untilId,
+      userId: [user._id],
+      blocking: true,
+    };
+
+    const relationships = await this.repositoryService
+      .findRelationships(filterQuery)
+      .sort({ followedUser: SortDirection.DESC })
+      .limit(maxResults)
+      .exec();
+
+    const userIds = relationships.map(({ followedUser }) => followedUser);
+
+    return this.getByCriteria(
+      user,
+      { _id: userIds },
+      {},
+      hasRelationshipExpansion
+    );
+  }
+
+  getByCriteria = async (
+    viewer: User,
+    query: FilterQuery<User>,
+    queryOptions?: CastcleQueryOptions,
+    hasRelationshipExpansion = false,
+    userFields?: UserField[]
+  ) => {
+    const { items: targetUsers, meta } = await this.getAllByCriteria(
+      query,
+      queryOptions
+    );
+
+    const users = await this.convertUsersToUserResponsesV2(
+      viewer,
+      targetUsers,
+      hasRelationshipExpansion,
+      userFields
+    );
+    return { users, meta };
+  };
+
+  getAllByCriteria = async (
+    query: FilterQuery<User>,
```

Comment:
```suggestion
  getAllByCriteria = async (
    query: FilterQuery<User>,
```
ตรงนี้ ส่ง { _id : userId } แนะนำให้ ใช้ class มารับดีกว่าครับ FilterQuery จะต้องไม่ ประกาศ any และ ลดการซ้ำซ้อน สุดท้ายฝั่ง  query: FilterQuery<User> ก็ไป prepare filterQuery อยู่ดี